src/Microsoft.Diagnostics.EventFlow.Core/Implementations/EventData.cs

```diff
@@ -184,6 +191,8 @@ public EventData DeepClone()
             other.Level = this.Level;
             other.ProviderName = this.ProviderName;
             other.Timestamp = this.Timestamp;
+            other.EventContext = this.EventContext;
```

Comment:
The problem with event context, as it is defined today, is that it is an "object". So we do not know its cloning semantics.

If the context is immutable, the current implementation is correct. If it is not, and it might be changing, then we should clone it, but because it is an object, we do not how.

One possibility is to check if the context implements IDeepCloneable. If it does, clone it. If not, do what current implementation does.